adobe_vipm/flows/validation/base.py

```diff
@@ -38,26 +30,24 @@ def validate_order(client, order):
     """
     try:
         has_errors = False
+        agreement_type = get_ordering_parameter(order, constants.PARAM_AGREEMENT_TYPE).get("value")
+
+        def validate_purchase_or_transfer(client, order):
+            if agreement_type == "Migrate" and is_transfer_validation_enabled(order):
+                return validate_transfer(client, order)
+            else:
+                return validate_purchase_order(client, order)
 
-        match order["type"]:
-            case constants.ORDER_TYPE_PURCHASE:
-                if is_transfer_order(order) and is_transfer_validation_enabled(
-                    order
-                ):
-                    adobe_client = get_adobe_client()
-                    order = populate_order_info(client, order)
-                    order = reset_ordering_parameters_error(order)
-                    order = reset_order_error(order)
-                    has_errors, order = validate_transfer(client, adobe_client, order)
-                else:
-                    has_errors, order = validate_purchase_order(client, order)
-            case constants.ORDER_TYPE_CHANGE:
-                has_errors, order = validate_change_order(client, order)
-            case constants.ORDER_TYPE_TERMINATION:
-                has_errors, order = validate_termination_order(client, order)
+        validators = {
+            constants.ORDER_TYPE_PURCHASE: validate_purchase_or_transfer,
```

Comment:
I think just `validate_purchase_order` is enough here. Purchase might be transfer or new customer. But we don't have "transfer" type of order